### ISSUE TITLE
aws_vpc_peering_connection catch-22

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#619]
_This issue was originally opened by @mobrien as hashicorp/terraform#12868. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version
v0.9.1

### Affected Resource(s)
aws_vpc_peering_connection
aws_vpc_peering_connection_accepter

### Terraform Configuration Files
```hcl

resource "aws_vpc_peering_connection" "bridge" {
    # provider      = "aws.requester"
    vpc_id          = "${var.requester_vpc}"
    peer_owner_id   = "${data.aws_caller_identity.accepter.account_id}"
    peer_vpc_id     = "${data.aws_vpc.accepter.id}"

    accepter {
        allow_remote_vpc_dns_resolution = true
    }
    requester {
        allow_remote_vpc_dns_resolution = true
    }
}

resource "aws_vpc_peering_connection_accepter" "bridge" {
    provider                  = "aws.accepter"
    vpc_peering_connection_id = "${aws_vpc_peering_connection.bridge.id}"
    auto_accept               = true
}
```

### Output
If doing cross-account peering, you cannot put auto_accept in the requester side. 
If a aws_vpc_peering_connection has an inline requester block, the apply fails because the accepter side has not yet accepted the connection.

### Expected
Should at least document that accepter blocks are not possible if doing cross account peering.
Further, you cannot use auto_accept on the requester side for cross account peering.

### Actual Behavior
Apply fails

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by jkroepke** [comment]
Somethink like an `aws_vpc_peering_connection_option` resource would be resolve this problem.

**Comment by FernandoMiguel** [comment]
I'm facing a similar issue.
although the request to the cross account is created, terraform fails to finish and push the state file to s3, preventing the cross account to read the peer_connection_id

**Comment by ewbankkit** [comment]
This has been fixed with the [`aws_vpc_peering_connection_options`](https://www.terraform.io/docs/providers/aws/r/vpc_peering_options.html) resource added via https://github.com/terraform-providers/terraform-provider-aws/pull/3909 and released in [version 1.17.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1170-may-02-2018).

**Comment by bflad** [comment]
Thanks @ewbankkit -- For anyone finding this please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
